index.js

```diff
@@ -1,23 +1,43 @@
-var extname = require('path').extname,
-    fs = require('fs'),
-    basename = require('path').basename,
-    browserify = require('browserify'),
-    watchify = require('watchify'),
-    debug = require('debug')('browserify-dev-middleware'),
-    _ = require('underscore');
+var PrettyError = require('pretty-error');
+var _ = require('underscore');
+var basename = require('path').basename;
+var beep = require('beepbeep')
+var browserify = require('browserify');
+var errorify = require('errorify');
+var debug = require('debug')('browserify-dev-middleware');
+var extname = require('path').extname;
+var fs = require('fs');
+var notifier = require('node-notifier')
+var watchify = require('watchify');
+
+var pe = new PrettyError();
 
 var watchers = {};
 var cached = {};
 
 var bundleAndCache = function(w, path) {
   w.bundle(function(err, src) {
     if (err) {
-      console.warn(err.message);
-      cached[path] = "alert(\"BROWSERIFY COMPILE ERROR (check your " +
-        "console for more details): " + err.message + "\");";
+      cached[path] = `
+        if (typeof window !== 'undefined') {
+          console.warn('Error bundling file: ${path}')
+          console.error('${err.message}')
+          alert("BROWSERIFY COMPILE ERROR (check your console for more details): ${err.message}");
```

Comment:
@craigspaeth - wondering about your thoughts on removing this. Since most people have a Notification Center of some kind this could probably go away? Having to close it is kinda annoying, but that said you can check that little checkbox...